### ISSUE TITLE
Help: def_roots() always returning Error: Unclassified

### ISSUE THREAD

**Issue by Navidem** [nrc/rls-analysis#139]
Hi, I am trying to write a pretty print for rsl data, I get an unexpected error returned. The repository is here:
https://github.com/Navidem/rls-pretty-print

What are we supposed to do to get it working in the current directory?
To regenerate the issue, here are the steps:

Steps: cargo run
Expected Result: Ok
Actual result: Error: Unclassified

```rust
    let analysis = rls_analysis::AnalysisHost::new(rls_analysis::Target::Debug);
    let _roots = analysis.def_roots()?;
```

Thanks in advance for your help.

**Comment by nrc** [comment]
You need to call one of the `reload` methods in order to read in the analysis information, e.g., `reload_with_blacklist` (https://github.com/nrc/rls-analysis/blob/master/src/lib.rs#L150). Use the rls-blacklist to get an actual blacklist. The path_prefix and base_dir arguments are a little confusing. The first one is Cargo's working directory (i.e., where you are running `cargo build`) and will contain the target directory. The second one is the root of the whole workspace - these two are usually the same, but can be different when you are building just one crate in a workspace project or where you have a complex build system (e.g., Servo).

**Comment by Navidem** [comment]
Thanks for the help!
Right now using `reload` I can get the roots. But when I run it in my project's directory, none of the roots are my own written crate! What am I missing?

**Comment by Navidem** [comment]
For future references, I was able to resolve the issue by first making a call to `cargo rustc`with `-Z save-analysis` option. By setting `CARGO_TARGET_DIR` environment variable to `target/rls` of my project, the analysis data are saved there and then can be `reload`ed.
